### PR TITLE
[move-prover] added support for adding trusted loop invariants

### PR DESCRIPTION
**Pull request by shazqadeer**
## Motivation

This PR adds trusted loop invariants into the generated Boogie code.  Some of the tests that had been disabled earlier because of lack of invariants have been enabled again.  I also added annotations for Vector::reverse, work that was earlier done by @kkmc, and uncommented some of the postconditions on Vector::remove that can now be verified.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Existing tests and couple more tests in verify_vector.move.

### CODE REVIEW

File: language/move-prover/tests/sources/functional/verify_vector.move [hunk 1]
```diff
@@ -97,22 +97,43 @@ module VerifyVector {
...
     // Reverses the order of the elements in the vector in place.
     fun verify_reverse<Element>(v: &mut vector<Element>) {
-        let len = Vector::length(v);
-        if (len == 0) return ();
+        let vlen = Vector::length(v);
+        if (vlen == 0) return ();
+
+        spec {
+            /// Initialize the old vector `old_v` ghost variable
+            assume forall k in 0..vlen: old_v<Element>[k] == v[k];
```
Comment (kkmc) [inline language/move-prover/tests/sources/functional/verify_vector.move:114]: Can probably replace it with `assume old_v<Element> == v`, although not important if we implement some version of old(.).

File: language/move-prover/tests/sources/functional/verify_vector.move [hunk 2]
```diff
@@ -231,21 +252,41 @@ module VerifyVector {
 
     // Remove the `i`th element E of the vector, shifting all subsequent elements
     // It is O(n) and preserves ordering
-    fun verify_remove<Element>(v: &mut vector<Element>, i: u64): Element {
-        let len = Vector::length(v);
+    fun verify_remove<Element>(v: &mut vector<Element>, j: u64): Element {
+        let vlen = Vector::length(v);
```
Comment (junkil-park) [inline language/move-prover/tests/sources/functional/verify_vector.move:256]: Just a note for later. We may need to adjust the name changes (i.e., "len" -> "vlen", and "i" -> "j") later when we add these specs into the actual stdlib vector file (`language/stdlib/modules/vector.move`).

Comment (kkmc) [reply language/move-prover/tests/sources/functional/verify_vector.move:256]: There is a related bug on using len #4355 .

**Pull request closed without merging**